src/Shared/ModelMethods.php

```diff
@@ -117,6 +117,22 @@ private function setProperty($propertyName, ...$value)
             } elseif ($actualType == 'string' && class_exists($expectedType) && is_array(json_decode($value, true))) {
                 $value = new $expectedType($value);
                 $actualType = get_class($value);
+            } elseif ($actualType == 'string' && $expectedType == 'boolean') {
+                if (in_array(strtolower($value), ['true', '1', 'yes', 'on'])) {
+                    $value = true;
+                    $actualType = gettype($value);
+                } elseif (in_array(strtolower($value), ['false', '0', 'no', 'off'])) {
+                    $value = false;
+                    $actualType = gettype($value);
+                }
+            } elseif ($actualType == 'integer' && $expectedType == 'boolean') {
+                if ($value === 1) {
+                    $value = true;
+                    $actualType = gettype($value);
+                } elseif ($value === 0) {
+                    $value = false;
+                    $actualType = gettype($value);
+                }
```

Comment:
This section could potentially be replaced with `filter_var()`. Some notes on the function - it became a part of default php starting in 5.2. The filter module can be installed on earlier versions. Can be seen working here: https://3v4l.org/2EiNN

Make sure to pass `FILTER_NULL_ON_FAILURE` as the third param otherwise anything that doesn't fall into the true values will return false. Empty string will also eval to false which isn't equivalent to what is here so may not meet requirements. 

See https://www.php.net/manual/en/filter.filters.validate.php for a list of validation filters including `FILTER_VALIDATE_BOOLEAN`.

```suggestion
            } elseif ($expectedType == 'boolean') {
                $filteredValue = filter_var($value, FILTER_VALIDATE_BOOLEAN, FILTER_NULL_ON_FAILURE);

                if (!is_null($filteredValue)) {
                    $value = $filteredValue;
                    $actualType = gettype($value);
                }
```